This patch is needed as the 'as' command shipped with MacOSX does not
accept the same commandlines as GNU as, that is instead shipped as gas.

Index: flac-1.1.2/configure.in
===================================================================
--- flac-1.1.2.orig/configure.in
+++ flac-1.1.2/configure.in
@@ -184,14 +184,17 @@ AC_DEFINE(FLAC__HAS_NASM)
 fi
 
 # only matters for PowerPC
-AC_CHECK_PROGS(AS, as)
-AC_CHECK_PROGS(GAS, gas)
-AM_CONDITIONAL(FLaC__HAS_AS, test -n "$AS")
-AM_CONDITIONAL(FLaC__HAS_GAS, test -n "$GAS")
-if test -n "$AS" ; then
+AC_CHECK_PROGS(AS, as, as)
+AC_CHECK_PROGS(AS, gas, gas)
+
+test "$AS" = "as" && as -v -o /dev/null < /dev/null | grep GNU >/dev/null && AS=gas
+
+AM_CONDITIONAL(FLaC__HAS_AS, test "$AS" = "as" )
+AM_CONDITIONAL(FLaC__HAS_GAS, test "$AS" = "gas" )
+if test "$AS" = "as" ; then
 AC_DEFINE(FLAC__HAS_AS)
 fi
-if test -n "$GAS" ; then
+if test "$AS" = "gas" ; then
 # funniest. macro. ever.
 AC_DEFINE(FLAC__HAS_GAS)
 fi
Index: flac-1.1.2/src/libFLAC/Makefile.am
===================================================================
--- flac-1.1.2.orig/src/libFLAC/Makefile.am
+++ flac-1.1.2/src/libFLAC/Makefile.am
@@ -59,12 +59,13 @@ if FLaC__CPU_PPC
 ARCH_SUBDIRS = ppc
 if FLaC__HAS_AS__TEMPORARILY_DISABLED
 libFLAC_la_LIBADD = ppc/as/libFLAC-asm.la
+LOCAL_EXTRA_LDFLAGS = "-Wl,-read_only_relocs,warning"
 else
 if FLaC__HAS_GAS__TEMPORARILY_DISABLED
 libFLAC_la_LIBADD = ppc/gas/libFLAC-asm.la
+LOCAL_EXTRA_LDFLAGS = ""
 endif
 endif
-LOCAL_EXTRA_LDFLAGS = "-Wl,-read_only_relocs,warning"
 endif
 endif
 
